Sources/ExperienceEvent.swift

```diff
@@ -24,31 +24,59 @@ public class ExperienceEvent: NSObject {
     /// Optional free-form query data associated with this event
     @objc public var query: [String: Any]?
 
+    /// Adobe Experience Platform datastream identifier, used to override the default datastream identifier set in the Edge Configuration
+    @objc public private(set) var datastreamIdOverride: String?
+
+    /// Adobe Experience Platform datastream configuration, used to override the datastream configuration
+    @objc public private(set) var datastreamConfigOverride: [String: Any]?
+
     /// Adobe Experience Platform dataset identifier, if not set the default dataset identifier set in the Edge Configuration is used
-    @objc public let datasetIdentifier: String?
+    @objc public private(set) var datasetIdentifier: String?
 
     /// Initialize an Experience Event with the provided event data
     /// - Parameters:
     ///   - xdm:  XDM formatted data for this event, passed as a raw XDM Schema data dictionary.
     ///   - data: Any free form data in a [String : Any] dictionary structure.
     ///   - datasetIdentifier: The Experience Platform dataset identifier where this event should be sent to; if not provided, the default dataset identifier set in the Edge configuration is used
-    @objc public init(xdm: [String: Any], data: [String: Any]? = nil, datasetIdentifier: String? = nil) {
+    @objc public convenience init(xdm: [String: Any], data: [String: Any]? = nil, datasetIdentifier: String? = nil) {
```

Comment:
Yes, this is intentional and we dont want the user to use all the three together, and also it will help in deprecating the datasetIdentifier which has to be migrated to datastreamConfigOverride route.